components/BackstoryQuestionaire.js

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useState } from "react";
+import styles from "../styles/Home.module.css";
+import Question from "./Question";
+import { backstoryQuestions } from "../data/backstoryQuestions";
+
+function formatRepsonse(questions, userAnswers, userOwnChoiceAnswers) {
+  const response = questions.map((questionData, index) => {
+    const updatedUserAnswers = userAnswers[index].map((updatedUserAnswer) =>
+      updatedUserAnswer.replace("Own choice", userOwnChoiceAnswers[index])
+    );
+    return { question: questionData.question, answers: updatedUserAnswers };
+  });
+
+  return response;
+}
+
+export default function BackstoryQuestionaire({
+  character,
+  setBackstoryExists,
+}) {
+  // Question useStates
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [currentQuestion, setCurrentQuestion] = useState(
+    backstoryQuestions[currentQuestionIndex]
+  );
+  // Answer useStates
+  const [answers, setAnswers] = useState(
+    new Array(backstoryQuestions.length).fill([])
+  );
+  const [ownChoiceAnswers, setOwnChoiceAnswers] = useState(
+    new Array(backstoryQuestions.length).fill("")
+  );
+  // Button useStates
+  const [nextButtonExists, setNextButtonExists] = useState(
+    currentQuestionIndex < backstoryQuestions.length - 1
+  );
+  const [previousButtonExists, setPreviousButtonExists] = useState(
+    currentQuestionIndex !== 0
+  );
+
+  // Updates Question, and button state
+  useEffect(() => {
+    const isNextButtonShown =
+      currentQuestionIndex < backstoryQuestions.length - 1;
+    const isPreviousButtonShown = currentQuestionIndex !== 0;
+
+    setCurrentQuestion(backstoryQuestions[currentQuestionIndex]);
+    setPreviousButtonExists(isPreviousButtonShown);
+    setNextButtonExists(isNextButtonShown);
+  }, [currentQuestionIndex]);
+
+  // Updates answers
+  function handleAnswer(answer) {
+    const newAnswers = [...answers];
+    if (newAnswers[currentQuestionIndex].includes(answer)) {
+      newAnswers[currentQuestionIndex] = newAnswers[
+        currentQuestionIndex
+      ].filter((a) => a !== answer);
+    } else {
+      newAnswers[currentQuestionIndex] = [
+        ...newAnswers[currentQuestionIndex],
+        answer,
+      ];
+    }
+    setAnswers(newAnswers);
+  }
+
+  // Updates own choice answers
+  function handleOwnChoice(answer) {
+    const newAnswers = [...ownChoiceAnswers];
+    newAnswers[currentQuestionIndex] = answer;
+    setOwnChoiceAnswers(newAnswers);
+  }
+
+  function handleNavigationButtonClick(e) {
+    e.preventDefault();
+
+    e.target.value === "next"
+      ? setCurrentQuestionIndex(currentQuestionIndex + 1)
+      : setCurrentQuestionIndex(currentQuestionIndex - 1);
```

Comment:
Another way to write this is:
```suggestion
setCurrentQuestionIndex(currentQuestionIndex + e.target.value === "next" ? 1 : -1)
```

Whichever you find more readable.

Reply:
Updating this breaks it completely for some reason, not sure why